src/lib.rs

```diff
@@ -284,12 +306,6 @@ impl<T> Node<T> {
 }
 
 impl NodeId {
-    /// Create a `NodeId` used for attempting to get `Node`s references from an
-    /// `Arena`.
-    pub fn new(index: usize) -> Self {
```

Comment:
We added this function because of #9 I understand your arguments but it is definitely in use.

Reply:
Ok, I understand some public constructor is necessary.
But then which do you think is preferable, `NodeId::new(index0: usize)` or `NodeId::from_index1(index1: NonZeroUsize)`?
I think `NonZeroUsize` version is better, because it would never panic and the fact "index is one-based" seems more explicit, but it is breaking change.

Reply:
Yeah I agree with you.

Reply:
I noticed that I can avoid breaking change:

```rust
impl NodeId {
    /// From zero-based index.
    /// Panics if `usize::max_value()` is passed.
    #[deprecated(since = "3.1.0", note="Use `NodeId::from_index0` or `NodeId::from_index1` instead")]
    pub fn new(index0: usize) -> Self { ... }
    /// From one-based index.
    pub fn from_index1(index1: NonZeroUsize) -> Self { ... }
}
```

I'm not sure deprecating `new()` is best, since `from_index1` leaks internal representation (which usually user doesn't concern), and it may reduce future compatibility.
However, without deprecating `new()`, users may continue using it without knowking `usize::max_varlue()`case.

Should `new()` be deprecated?
Or, should it be changed to `fn new(index1: NonZeroUsize)` and cause breaking change?

Reply:
I had a short recap of the things and tend to revise my previous comment. If we keep the `NodeID::new(index: usize)` version which can panic the end user does not have to change already existing code.

Another approach would be to go for `NodeID::new(index: usize) -> Fallible<Self>`...

Reply:
I force-pushed the latest commit (https://github.com/saschagrunert/indextree/pull/20/commits/0d1e2835b370db9738d5a87f8770e8d6e67270b5).
This keeps `pub fn NodeId::new(usize)` compatible with the latest released version, and adds new `pub fn NodeId::from_index1(NonZeroUsize)`.
These are non-breaking change.

Is this OK?
(I think `from_index1` can be named more clearly, such as `from_one_based_index` or something like that.)

Reply:
Yeah, can you merge the two impl NodeId blocks into one? We could cal it `from_non_zero_usize`, couldn’t we? :)